features/netsocket/NetworkStack.h

```diff
@@ -115,6 +132,30 @@ class NetworkStack: public DNS {
     virtual nsapi_value_or_error_t gethostbyname_async(const char *host, hostbyname_cb_t callback, nsapi_version_t version = NSAPI_UNSPEC,
                                                        const char *interface_name = NULL);
 
+    /** Translates a hostname to the multiple IP addresses (asynchronous)
+     *
+     *  The hostname may be either a domain name or an IP address. If the
+     *  hostname is an IP address, no network transactions will be performed.
+     *
+     *  If no stack-specific DNS resolution is provided, the hostname
+     *  will be resolve using a UDP socket on the stack.
+     *
+     *  Call is non-blocking. Result of the DNS operation is returned by the callback.
+     *  If this function returns failure, callback will not be called. In case result
+     *  is success (IP address was found from DNS cache), callback will be called
+     *  before function returns.
```

Comment:
This is a bit missleading.

It now sounds like the callback will always be called before the call returns, which is not true. To my understand, it MAY call the callback before returning IF the result was found from the cache. Otherwise it returns immediately and query will be send in the background.

Reply:
Also, you need to now specify that callback `status` parameter contains number of results returned.